lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Transactions.hs

```diff
@@ -1035,6 +1036,28 @@ spec = describe "SHELLEY_TRANSACTIONS" $ do
             , expectField #expiresAt (`shouldSatisfy` isJust)
             ]
 
+    it "TRANSFORGE_MINT_01 - Mint tokens" $ \ctx -> runResourceT $ do
+      w <- emptyWallet ctx
+
+      addrs <- listAddresses @n ctx w
+      let destination = (addrs !! 1) ^. #id
+      let payload = Json [json|{
+                              "forge_payments": [{
+                                "address": #{destination},
+                                "amount": {
+                                    "quantity": 5
+                                }
+                              }],
+                              "passphrase": #{fixturePassphrase},
+                              "asset_name": "aaaa"
```

Comment:
I guess that's still a draft, but just wanted to ask: are we going to provide parameter for policy_id or policy script?

Reply:
That's a really good discussion to have. I'm sure @rvl will have some thoughts.

I think the main thing that the wallet offers in this regard is the management and recovery of keys. If the user provides their own policy, they will also have to provide us their minting keys, so we can sign the transaction. But in this case we can't guarantee recovery of these keys with the seed phrase. I think if a user wants to manage their own keys, using cardano-cli is the best choice and offers the least amount of confusion.

The plan was to allow the user to provide the asset name, but generate the minting keys under a key derivation branch of the wallet, so the minting keys can be recovered and managed by the wallet. The policy is generated from that key and we'd return to the user the policy_id generated (or some other identifier), so that they can ask the wallet to mint more of the same assets, or new assets under the same policy, or to burn existing assets.

Reply:
I'm only considering the most basic of policy scripts, the "transactions must be signed by this key". I think more complex policy scripts are out-of-scope for this task but might be worth thinking about now.

Reply:
Ok, cool. I was kind of referring to acceptance criteria from the ticket:

> Acceptance Criteria
> 
>  - I need to be able to generate a token,
> by specifying a token policy and an amount
>  - Script needs to be provided by the user
>  - Forging amount needs to be provided by the user

Perhaps they need to be updated then / double checked with the PO?